tests/nn/beam_search_test.py

```diff
@@ -277,6 +277,26 @@ def test_top_k_search(self):
         # log_probs should be shape `(batch_size, beam_size, max_predicted_length)`.
         assert list(log_probs.size()) == [batch_size, beam_size]
 
+    def test_stochastic_beam_search(self):
+        initial_predictions = torch.tensor([0] * 5)
+        beam_size = 3
+        take_step = take_step_with_timestep
+
+        top_k, log_probs = BeamSearch.stochastic_beam_search(
+            self.end_index, beam_size=beam_size
+        ).search(initial_predictions, {}, take_step)
+
+        beam_size = beam_size or 1
+        batch_size = 5
+
+        # top_p should be shape `(batch_size, beam_size, max_predicted_length)`.
+        assert list(top_k.size())[:-1] == [batch_size, beam_size]
+
+        assert ((0 <= top_k) & (top_k <= 5)).all()
+
+        # log_probs should be shape `(batch_size, beam_size, max_predicted_length)`.
+        assert list(log_probs.size()) == [batch_size, beam_size]
```

Comment:
We should be able to verify exactly what `log_probs` should be given `top_k`, which would be good to do in this test. If that's too difficult though, we could have a similar test using a dummy sampler that does the same thing as deterministic beam search. Then we'd just have to check that the `log_probs` result is the same when using the dummy sampler vs using no sampler.